neuror/cut_plane/detection.py

```diff
@@ -157,6 +158,27 @@ def find(cls, neuron, bin_width=3,
             best_plane.coefs[3] = coef_d
         return CutPlane(best_plane.coefs, best_plane.upward, neuron, best_plane.bin_width)
 
+    @classmethod
+    def find_legacy(cls, neuron, axis):
+        '''Find the cut points according to the legacy algorithm
+
+        As implemented in:
+        https://bbpcode.epfl.ch/source/xref/platform/BlueRepairSDK/BlueRepairSDK/src/repair.cpp#263
+        '''
+        if not isinstance(neuron, Neuron):
+            neuron = load_neuron(neuron)
+
+        cut_leaves, side = cut_detect(neuron, axis)
+
+        plane = CutPlane([int(axis.upper() == 'X'), int(axis.upper() == 'Y'),
```

Comment:
I am always puzzle by this design and can t figure out if this is a good one or not. 

Having a class method returning an explicit constructor messes up the inheritance but in another hand it prevents the classmethod to fail if the daughter class have a different constructor ... 

Any thoughts ?

Reply:
I think you are right, I should have used `cls`.
I think it's fine to assume that the derived class have the same ctor. What else can you do anyway ?

Reply:
yeah indeed. 
But with this stuff, I am starting to think the classmethod has a strange behavior in general.